src/common/tests/lru.c

```diff
@@ -127,6 +127,12 @@ main(int argc, char **argv)
 	if (rc)
 		D_ASSERTF(0, "Error in creating lru cache\n");
 
+	if (num_keys < 0 || num_keys > INT_MAX) {
```

Comment:
I think this ` > INT_MAX` check would never be true, since `num_keys` is an `int` type, so can never hold a value greater than `INT_MAX`.
For example, if a user passes `INT_MAX + 1 = 2147483648`, it will overflow to `-2147483648`.
If we want to check for overflow, `strtol` actually returns a `long int`:
https://linux.die.net/man/3/strtol
But I'm not familiar enough with this code to know what the intention is

Reply:
@daltonbohning the ticket mentions a fix as checking a lower and upper bound for num_keys, but I wasn't aware there was an upper bound for the number of keys, but here a value could definitely overflow since it is using strtol and casting to an int. Maybe I have to create another variable that holds a long int on the strtol conversion, check that it is not greater than INT_MAX before casting to an int?

Reply:
@dsikich That's what I would think. `strtol `allows you to specify the number the characters to convert, so maybe you can use 11 instead of 10 to also avoid overflowing the `long int`. But TBH I've not used this function before

Reply:
@daltonbohning I think I am going to just make csize and num_keys a long int and then check if num_keys < 0 which should catch an overflow of long int if it occurs.

Reply:
@dsikich But does this value NEED to be an int when used? I.e. would making it a long int cause overflow somewhere else? If it's a long int, we probably still need the upper bounds check to make sure it doesn't overflow an int value

Reply:
@daltonbohning hmm not familiar enough with this code to know that, I suppose leaving the check in for INT_MAX is best in that case.